common/src/main/java/tachyon/security/PlainSaslServerProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security;
+
+import java.security.Provider;
+import java.util.Map;
+
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.sasl.SaslException;
+import javax.security.sasl.SaslServer;
+import javax.security.sasl.SaslServerFactory;
+
+public class PlainSaslServerProvider extends Provider {
+  public static final String TACHYON_PLAIN_NAME = "TachyonSaslPlain";
+
+  public PlainSaslServerProvider() {
+    super(TACHYON_PLAIN_NAME, 1.0, "Tachyon Plain SASL provider");
+    put("SaslServerFactory.PLAIN", SaslPlainServerFactory.class.getName());
+  }
+
+  public static class SaslPlainServerFactory implements SaslServerFactory {
```

Comment:
NIT: I am a bit curious here, because the enclosing class is called `PlainSaslServerProvider` while this factory class is called `SaslPlainServerFactory`.  Do you have any particular reason for the inconsistent naming (e.g., `SaslPlainServerFactory` will be more consistent with a lot more new coming classes)?